test/runner/lib/executor.py

```diff
@@ -796,6 +796,19 @@ def command_sanity_validate_modules(args, targets):
     if skip_paths:
         cmd += ['--exclude', '^(%s)' % '|'.join(skip_paths)]
 
+    # Find fork_branch
+    git = Git(args)
+    try:
+        changes = ShippableChanges(args, git) if is_shippable() else LocalChanges(args, git)
+    except ApplicationError:
```

Comment:
Several things concern me about this approach:

- Change detection is run twice on Shippable and also locally if the `--changed` option was used.
- When run locally, `devel` may be ahead/behind the current branch (aside from intentional changes).
- Local change detection has several bugs which make it fail often. This is related the previous point.
- Any kind of change detection failure results in the `devel` branch being used.

As much as I'd like to see this be able to be run seamlessly locally and on Shippable, I think we should just start with supporting Shippable correctly, until we have a better solution for handling local change detection (not just for this, but for all of ansible-test).

I'm thinking we can do something along the lines of:

- Make the `--base-branch` checking optional. If not specified, don't perform that part of module validation.
- If `is_shippable()` then pass `os.environ['BASE_BRANCH']` as the base branch.
- If not `is_shippable()` then show a warning about skipping the additional module validation test.

I generally don't like making the tests Shippable aware, but in this case it's necessary.